src/classes/action/movie.vala

```diff
@@ -498,10 +526,35 @@ namespace pdfpc {
             cr.scale(this.scalex, -this.scaley);
 
             this.draw_seek_bar(cr, timestamp);
-        }
 
+            // if a stop time is defined, stop there (but still let
+            // the user manually seek *after* this timestamp)
+            if (this.stoptime != 0 && 
+                this.stoptime * Gst.SECOND < timestamp &&
+                timestamp < (this.stoptime + 0.2) * Gst.SECOND) {
+                if (this.loop) {
+                    this.pipeline.seek_simple(Gst.Format.TIME, Gst.SeekFlags.FLUSH, this.starttime * Gst.SECOND);
+                } else {
+                    // Can't seek to beginning w/o updating output, so mark to seek later
+                    this.eos = true;
+                    this.pause();
+                }
+            }
+
+        }
+        
         private void draw_seek_bar(Cairo.Context cr, uint64 timestamp) {
+            double start = 1.0*this.starttime*Gst.SECOND / this.duration;
+            double stop = 1.0*this.stoptime*Gst.SECOND / this.duration;
```

Comment:
I'd prefer an explicit cast to double, rather than the implicit one caused by multiplying by 1.0.  Also, spaces around the * operators.